### PR TITLE
NeMo-UX: use input_ids instead of tokens in HfAutoModelForCausalLM

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

HfAutoModelForCausalLM's forward used input named `tokens` but the underlying model was using `input_ids` coincidenting with mcore's `input_ids`. -> renamed param to `input_ids` to have a unified interface.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py [hunk 1]
```diff
@@ -82,41 +84,31 @@ def configure_model(self):
             self.model = AutoModelForCausalLM.from_config(config, trust_remote_code=self.trust_remote_code)
         self.model.train()
 
-    def forward(self, input_ids, attention_mask=None, labels=None, loss_mask=None):
+    def forward(
+        self,
+        input_ids: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+        position_ids: torch.Tensor = None,
```
Comment (oyilmaz-nvidia) [inline nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py:91]: Can we pass this position_ids to the model as well?

**Review by oyilmaz-nvidia — CHANGES_REQUESTED** [review]

File: nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py [hunk 2]
```diff
@@ -82,41 +84,31 @@ def configure_model(self):
...
-        labels = batch['labels']
-        loss_mask = batch.get('loss_mask', None)
-        output = self.forward(
-            input_ids=tokens,
-            labels=labels,
-            loss_mask=loss_mask,
-        )
-
-        loss = output.loss
+        loss = self.forward(**batch)
```
Comment (oyilmaz-nvidia) [inline nemo/collections/llm/gpt/model/hf_auto_model_for_causal_lm.py:106]: Can you please update the https://github.com/NVIDIA/NeMo/blob/akoumparouli/nemo_ux_fix_hf_auto_model_param_names/examples/llm/sft/hf.py as well?
